cli/wallet/wallet.go

```diff
@@ -542,27 +557,61 @@ func importMultisig(ctx *cli.Context) error {
 		}
 	}
 
-	var label *string
 	if ctx.IsSet("name") {
 		l := ctx.String("name")
 		label = &l
 	}
-	acc, err := newAccountFromWIF(ctx.App.Writer, ctx.String("wif"), wall.Scrypt, label, pass)
-	if err != nil {
-		return cli.NewExitError(err, 1)
-	}
 
-	if err := acc.ConvertMultisig(m, pubs); err != nil {
-		return cli.NewExitError(err, 1)
+	for _, pub := range pubs {
+		match := findAccountsByPublicKey(wall, pub)
+		if len(match) > 0 {
+			matchingAccounts = append(matchingAccounts, match...)
+			accEncPub = pub
+		}
 	}
 
-	if err := addAccountAndSave(wall, acc); err != nil {
-		return cli.NewExitError(err, 1)
+	if len(matchingAccounts) == 1 {
+		acc = matchingAccounts[0]
+		err = acc.ConvertMultisigEncrypted(accEncPub, m, pubs)
+		if err != nil {
+			return cli.NewExitError(err, 1)
+		}
+		wall.AddAccount(acc)
+		if err := wall.Save(); err != nil {
+			return cli.NewExitError(err, 1)
+		}
+		return nil
 	}
+	if ctx.IsSet("wif") {
```

Comment:
`if !ctx.IsSet("wif") {` then immediately return exit error. And then no extra `else` is needed.

Reply:
Not resolved. Check the WIF firstly and return an error if it's not set. It's just a better coding style, because further you won't have nested IF statement.